src/eureka/lib/disk.py

```diff
@@ -57,3 +58,65 @@ def disk(r, ctr, size, status=False):
     if status:
         ret = ret, retstatus
     return ret
+
+
+def hex(r, ctr, size, status=False):
+    """
+    This function returns a byte array containing a hexagon.
+
+    The hexagon is centered at (ctr[0], ctr[1]), and is circumscribed by a 
+    circle of radius r. The array is (size[0], size[1]) in size and has byte 
+    type. Pixel values of 1 indicate that the center of a pixel is within r 
+    of (ctr[0], ctr[1]). Pixel values of 0 indicate the opposite. The center 
+    of each pixel is the integer position of that pixel.
+
+    Parameters
+    ----------
+    r : float
+        The radius of the circle circumscribing the hexagon.
+    ctr : tuple, list, or array
+        The x,y position of the center of the hexagon, 2-element vector.
+    size : tuple, list, or array
+        The x,y size of the output array, 2-element vector.
+    status : bool; optional
+        If True, return the status optional output.
+
+    Returns
+    -------
+    ret : ndarray
+        A boolean array where False if outside the hexagon or True if inside
+        the hexagon.
+    retstatus : int; optional
+        Set to 1 if any part of the hexagon is outside the image boundaries.
+        Only returned if status==True.
+
+    Notes
+    -----
+    History:
+        - 2024-06-05: Yoni Brande, jbrande@ku.edu
+        Initial version, adapted from a snippet by Ian Crossfield
+    """
+
+    # check if hex is off image (same check as disk, for now)
+    retstatus = int(ctr[0] - r < 0 or ctr[0] + r > size[0]-1 or
+                    ctr[1] - r < 0 or ctr[1] + r > size[1]-1)
+    
+    # make hexagon, oriented like the JWST mirrors (corner vertex at 12:00)
+    yvert = ctr[1] + r*np.cos(2*np.pi*np.arange(6)/6)
+    xvert = ctr[0] - r*np.sin(2*np.pi*np.arange(6)/6)
```

Comment:
Just want to double-check that you don't need to do `np.arange(7)` like you did for the S3 plots - it's certainly possible this is fine as is, just want to be sure

Reply:
Yep, the extra vertex is just for closing path patches when plotting, the actual pixel mask construction doesn't need it. See [the API for this](https://matplotlib.org/stable/api/path_api.html). It's a little confusing, because the Polygon patch _doesn't_ [work that way](https://matplotlib.org/stable/api/_as_gen/matplotlib.patches.Polygon.html).